test/libraries/Hooks.t.sol

```diff
@@ -741,10 +746,13 @@ contract HooksTest is Test, Deployers, GasSnapshot {
         assertTrue(hookAddr.hasPermission(Hooks.AFTER_DONATE_FLAG));
     }
 
-    function test_validateHookAddress_failsAllHooks(uint152 addr, uint8 mask) public {
-        uint160 preAddr = uint160(uint256(addr));
-        vm.assume(mask != 0xff8);
-        IHooks hookAddr = IHooks(address(uint160(preAddr) | (uint160(mask) << 151)));
+    function test_validateHookAddress_failsAllHooks(uint160 addr, uint16 mask) public {
+        uint160 preAddr = uint160(uint256(addr) & clearAllHookPermisssionsMask);
+        // Set the upper `hooksPermissionCount` number of bits to get the full mask in uint16.
+        uint16 allHooksMask = (~uint16(0) << uint16(16 - hookPermissionCount));
```

Comment:
This isnt really test-specified so no need to execute it every time the test runs. Feels like it could sit globally with `clearAllHookPermisssionsMask`?

Reply:
actually on that note why is `clearAllHookPermisssionsMask` in the setup function - can't these both just be immutables that are calculated once in the contract and never again?

Reply:
yeah sure, its just calculated `clearAllHookPermisssionsMask` based on numHookPermissions. I added numHookPermissions so the tests would be super easy to update as we added more permissions. I'd like to keep it that way. But this var and the allHooksMask can be set at the top level

Reply:
To close the loop here `hookPermissionCount` and `clearAllHookPermisssionsMask` are all declared in Deployers.sol (fixed in sep PR). We only really use allHooksMask here once in this file and no where else so I left it.